### PR TITLE
Support chat history persistence in AutoGPT

### PR DESCRIPTION
**Pull request by alonroth**
**Short Description**
Added a new argument to AutoGPT class which allows to persist the chat history to a file.

**Changes**
1. Removed the `self.full_message_history: List[BaseMessage] = []`
2. Replaced it with `chat_history_memory` which can take any subclasses of `BaseChatMessageHistory`

### CODE REVIEW

File: langchain/experimental/autonomous_agents/autogpt/agent.py [hunk 1]
```diff
@@ -37,15 +37,20 @@ def __init__(
...
     ):
         self.ai_name = ai_name
         self.memory = memory
-        self.full_message_history: List[BaseMessage] = []
         self.next_action_count = 0
         self.chain = chain
         self.output_parser = output_parser
         self.tools = tools
         self.feedback_tool = feedback_tool
+        self.chat_history_memory = (
```
Comment (hwchase17) [inline langchain/experimental/autonomous_agents/autogpt/agent.py:49]: lets just change this so users can pass in a memory object

Comment (alonroth) [reply langchain/experimental/autonomous_agents/autogpt/agent.py:49]: make sense :) will do

Comment (alonroth) [reply langchain/experimental/autonomous_agents/autogpt/agent.py]: Done + updated the relevant doc

**Comment by alonroth** [comment]
@hwchase17
When should this PR merged?
Thanks!

**Review by hwchase17 — APPROVED** [review]
thanks!

**Pull request merged**